SharedPackages/BrowserServicesKit/Sources/NetworkProtection/Status/VPNEnabledObserver/VPNEnabledObserverThroughSession.swift

```diff
@@ -0,0 +1,194 @@
+//
+//  VPNEnabledObserverThroughSession.swift
+//
+//  Copyright © 2025 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Combine
+import Foundation
+import NetworkExtension
+import NotificationCenter
+import Common
+import os.log
+
+public class VPNEnabledObserverThroughSession: VPNEnabledObserver {
+
+    public var isVPNEnabled: Bool {
+        subject.value
+    }
+
+    public lazy var publisher: AnyPublisher<Bool, Never> = subject.eraseToAnyPublisher()
+    private let subject: CurrentValueSubject<Bool, Never>
+
+    private let tunnelSessionProvider: TunnelSessionProvider
+    private let extensionResolver: VPNExtensionResolving
+
+    // MARK: - Notifications
+
+    private let notificationCenter: NotificationCenter
+    private let platformSnoozeTimingStore: NetworkProtectionSnoozeTimingStore
+    private let platformNotificationCenter: NotificationCenter
+    private let platformDidWakeNotification: Notification.Name
+    private var cancellables = Set<AnyCancellable>()
+
+    // MARK: - Initialization
+
+    public init(tunnelSessionProvider: TunnelSessionProvider,
+                extensionResolver: VPNExtensionResolving,
+                notificationCenter: NotificationCenter = .default,
+                platformSnoozeTimingStore: NetworkProtectionSnoozeTimingStore,
+                platformNotificationCenter: NotificationCenter,
+                platformDidWakeNotification: Notification.Name) {
+
+        self.extensionResolver = extensionResolver
+        self.notificationCenter = notificationCenter
+        self.platformSnoozeTimingStore = platformSnoozeTimingStore
+        self.platformNotificationCenter = platformNotificationCenter
+        self.platformDidWakeNotification = platformDidWakeNotification
+        self.tunnelSessionProvider = tunnelSessionProvider
+
+        // Unfortunately we can't set the initial value from real data without making the init
+        // async, so for now we'll be content to allow this to be false and spawn a task to
+        // update it.
+        subject = CurrentValueSubject<Bool, Never>(false)
+
+        Task { [tunnelSessionProvider] in
+            guard let activeSession = await tunnelSessionProvider.activeSession() else {
+                return
+            }
+
+            updateSubject(with: activeSession)
+        }
```

Comment:
Kinda hackish.  I didn't want to force a lot of changes by making this initializer `async`.